src/subgraph/zns/actions/getMostRecentSubdomainsById.ts

```diff
@@ -32,23 +38,13 @@ export const getMostRecentSubdomainsById = async <T>(
       { parent: domainId, count: yetUnreceived, startIndex: skip }
     );
 
-    const queriedDomains = queryResult.data.domains;
+    queriedDomains = queryResult.data.domains;
     for (const domain of queriedDomains) {
       subDomains.push(convertDomainDtoToDomain(domain));
     }
-
-    /**
-     * We will only get back up to `queryCount` # of domains
-     * So if we get that many there's probably more domains we need
-     * to fetch. If we got back less, we can stop querying
-     */
-     yetUnreceived -= queriedDomains.length;
-    if (queriedDomains.length < count || yetUnreceived <= 0) {
-      break;
-    }
+    yetUnreceived -= queriedDomains.length;
     skip = queriedDomains[queriedDomains.length - 1].indexId;
-  }
-
+  } while (yetUnreceived > 0 && queriedDomains.length != 0);
```

Comment:
This logic is slightly different. `&& queryDomains.length === count` would be the equivalent

Reply:
Fixed